scala/data_structures/trie.sc

```diff
@@ -0,0 +1,97 @@
+package data_structures
+
+final case class Trie(
+  private val children: Array[Option[Trie]],
+  private val isFinal: Boolean
+) {
+
+  private def getIndexesFromString: String => Seq[Int] =
+    _
+      .toCharArray
+      .map(_.toLower.charValue - 97) // 'a' is 97, 'b' is 98, etc
```

Comment:
The right operand of `-` should be `'a'` so that the expression is more readable without relying on comments.
Character literals evaluate to their ASCII value.